lib/block-supports/layout.php

```diff
@@ -371,16 +371,24 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		return (string) $content;
 	}
 
-	$block_gap              = gutenberg_get_global_settings( array( 'spacing', 'blockGap' ) );
-	$global_layout_settings = gutenberg_get_global_settings( array( 'layout' ) );
-	$has_block_gap_support  = isset( $block_gap ) ? null !== $block_gap : false;
-	$default_block_layout   = _wp_array_get( $block_type->supports, array( '__experimentalLayout', 'default' ), array() );
-	$used_layout            = isset( $block['attrs']['layout'] ) ? $block['attrs']['layout'] : $default_block_layout;
-
-	if ( isset( $used_layout['inherit'] ) && $used_layout['inherit'] ) {
-		if ( ! $global_layout_settings ) {
-			return $block_content;
-		}
+	static $static_information_computed   = false;
```

Comment:
Can we add a comment here to describe why we cache this data at render time, so future readers understand _why_ this is set up in this way?

Reply:
How is the cache invalidation?

Reply:
Looking at this code closer, I think we should remove this cache all together. Caching of global settings should be done here - https://github.com/WordPress/gutenberg/pull/45372. Cache invalidation should also be done here. That way we can call `gutenberg_get_global_settings` without worrying parsing again. 

The worry here is double caching and cache invalidation.

Reply:
@spacedmonkey How is cache invalidation relevant for this change? I'm not sure I understand.

Since this PR uses a simple "variable cache" (i.e. in PHP runtime memory), I wouldn't think cache invalidation for this matters (since it's not using `WP_Object_Cache` and the data here is static).

Reply:
@felixarntz What if a filter or register theme support changes something in the runtime? How is this runtime cache invalidated?